cdc/log.cc

```diff
@@ -543,6 +543,14 @@ static schema_ptr create_log_schema(const schema& s, std::optional<utils::UUID>
     auto add_columns = [&] (const schema::const_iterator_range_type& columns, bool is_data_col = false) {
         for (const auto& column : columns) {
             auto type = column.type;
+            if (type->get_kind() == abstract_type::kind::empty) {
+                auto& columns = s.all_columns();
+                auto regular_columns_count = std::count_if(columns.begin(), columns.end(), [](const column_definition& c) {
+                    return c.kind == column_kind::regular_column;
+                });
+                assert(s.is_dense() && regular_columns_count == 1);
```

Comment:
assert() is no longer fashionable - please use on_internal_error. See https://github.com/scylladb/scylla/issues/7871

Reply:
done.